operators/multitenancy-operator/MultitenancyService.js

```diff
@@ -0,0 +1,112 @@
+    } else if (operation.type === CONST.OPERATION_TYPE.DELETE || operation.type === CONST.OPERATION_TYPE.UPDATE) {
+      return Promise.try(() => {
+        this.deploymentName = operation.parameters.deploymentName;
+        logger.debug(`Dedicated instance deployment name is '${this.deploymentName}'...`);
+      });
+    }
+  }
+
+  create() {
+    const operation = {
+      type: CONST.OPERATION_TYPE.CREATE
+    };
+    return this.initialize(operation)
+      .tap(() => logger.info(`Creating resource '${this.guid}' of type '${this.resourceType}' on dedicated deployment '${this.deploymentName}'...`))
+      .then(() => this.storeDedicatedInstanceDeploymentName(this.deploymentName, this.guid))
+      .then(() => this.director.getDeploymentIps(this.deploymentName))
+      .then(ips => this.agent.createTenant(ips, this.guid, this.parameters))
+      .tap(() => logger.info(`+-> Created resource'${this.guid}' of type '${this.resourceType}' on dedicated deployment  '${this.deploymentName}'`));
+  }
+
+  update(changedOptions) {
+    const operation = {
+      type: 'update',
+      parameters: {
+        deploymentName: changedOptions.operatorMetadata.dedicatedInstanceDeploymentName
+      }
+    };
+    return this.initialize(operation)
+      .tap(() => logger.info(`Updating resource'${this.guid}' of type '${this.resourceType}' on dedicated deployment '${this.deploymentName}'...`))
+      .then(() => this.director.getDeploymentIps(this.deploymentName))
+      .then(ips => this.agent.updateTenant(ips, this.guid, this.parameters))
+      .tap(() => logger.info(`+-> Updated resource '${this.guid}' of type '${this.resourceType}' on dedicated deployment '${this.deploymentName}'`));
+  }
+
+  delete(changedOptions) {
+    const operation = {
+      type: 'delete',
+      parameters: {
+        deploymentName: changedOptions.operatorMetadata.dedicatedInstanceDeploymentName
+      }
+    };
+    let instanceDeleted = true;
+    return this.initialize(operation)
+      .tap(() => logger.info(`Deleting resource '${this.guid}' of type '${this.resourceType}' on dedicated deployment '${this.deploymentName}'...`))
+      .then(() => this.director.getDeploymentIps(this.deploymentName))
+      .tap(() => instanceDeleted = false)
+      .then(ips => Promise.all([this.agent.deleteTenant(ips, this.guid)]))
+      .tap(() => logger.info(`+-> Deleted resource '${this.guid}' of type '${this.resourceType}' on dedicated deployment '${this.deploymentName}'`))
+      .catch(NotFound, () => {
+        throw new Gone(this.guid);
+      });
+  }
+
+
+  storeDedicatedInstanceDeploymentName() {
+    return eventmesh.apiServerClient.patchResource({
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+      resourceType: this.resourceType,
+      resourceId: this.guid,
+      operatorMetadata: {
+        dedicatedInstanceDeploymentName: this.deploymentName
+      }
+    });
+  }
+
+  static createInstance(instanceId, options, resourceType) {
```

Comment:
If there is no real need for a createInstance method, we can take this out. I believe this class is no longer a singleton.

Reply:
Thanks for the suggestion !
Let's take this change in next sprint.